.circleci/real_config.yml

```diff
@@ -112,85 +112,77 @@ commands:
       - run: git submodule sync
       - run: git submodule update --init
 
-  skip-if-docs-only:
+  add-and-fetch-upstream:
     steps:
       - run: git remote add upstream https://github.com/determined-ai/determined
       - run: tools/scripts/retry.sh git fetch upstream
-      - run:
-          name: check for docs-only changes
-          command: |
-            MERGE_BASE="$(git merge-base upstream/main HEAD)"
-            DIFF_DIRS="$(git diff-tree --no-commit-id --name-only "$MERGE_BASE" HEAD)"
-            if [ "$DIFF_DIRS" = "docs" ] ; then
-              echo "docs-only change detected, halting job now"
-              circleci-agent step halt
-            else
-              echo "detected non-docs-only changes: $DIFF_DIRS"
-            fi
+
+  skip-if-dir:
+    # TODO review the commands
+    # We can combine them into one command and use bash to evaluate the parameter
+    # We should avoid *"wildcards like that with no delimiters"*
+    parameters:
+      dir:
+        type: string
+      negate:
```

Comment:
I would much prefer for the two cases to stay entirely separate. The current interface suggests that flipping the `negate` parameter always changes whether the skip occurs, which is not true. Also, the two cases don't share any code, and merging the two execution paths only to immediately separate them does nothing but add extra lines and mental overhead. (Though even if you did make the code overlap some, I think the first point would mean the cases should still stay separate.)